app/code/VConnect/Blog/Service/PostsProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\Blog\Service;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\FilterBuilder;
+use VConnect\Blog\Api\PostRepositoryInterface;
+use Magento\Framework\Api\SortOrderBuilder;
+
+class PostsProvider
+{
+    /**
+     * PostsProvider constructor.
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param FilterBuilder $filterBuilder
+     * @param SortOrderBuilder $sortOrderBuilder
+     * @param PostRepositoryInterface $postRepository
+     */
+    public function __construct(
+        private SearchCriteriaBuilder $searchCriteriaBuilder,
+        private FilterBuilder $filterBuilder,
+        private SortOrderBuilder $sortOrderBuilder,
+        private PostRepositoryInterface $postRepository
+    ) {}
+
+    /**
+     * @return \VConnect\Blog\Api\Data\PostInterface[]
+     */
+    public function getPosts(): array
+    {
+        /** @var \Magento\Framework\Api\Filter $filter1 */
+        $filter1 = $this->filterBuilder
```

Comment:
what is the reason to write $filter1? As i can see you have only one filter and you could simply use \Magento\Framework\Api\SearchCriteriaBuilder::addFilter

It uses filter builder inside and adds it depends on settings you set.